### DISCUSSION TITLE
[mini.base16]

### DISCUSSION THREAD

**Discussion by alezunino** [nvim-mini/mini.nvim#451] [Q&A] [answered]
Hi,
I am trying to create a colorscheme that follows my terminal colors. The idea is similar to https://github.com/noahfrederick/vim-noctu, but including nvim hl groups. I suspect mini.base16 can be used to build such a colorscheme by initializing it with the current 16 ANSI terminal colors. How can I achieve that?

**Comment by echasnovski** [comment, marked as answer]
> Hi, I am trying to create a colorscheme that follows my terminal colors. The idea is similar to https://github.com/noahfrederick/vim-noctu, but including nvim hl groups. I suspect mini.base16 can be used to build such a colorscheme by initializing it with the current 16 ANSI terminal colors. How can I achieve that?

Yes, indeed, 'mini.base16' can do that, but to a certain extent. Its 16 colors don't follow the usual agreement (like [this one](https://en.wikipedia.org/wiki/ANSI_escape_code#3-bit_and_4-bit)) but rather [its own guideline](https://github.com/chriskempson/base16/blob/main/styling.md).

Besides that, I am not aware of the way to get terminal colors from inside Neovim in automated fashion.
